File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int i = 0, j = 0;
    int size = brown + yellow;
    for(int i = 3; i <= size/2; i++){
        j = size / i;
        
        if((i-2)*(j-2) == yellow){
            answer.push_back(i);
            answer.push_back(j);
            break;
        }
    }
    int temp;
    if(answer[0] < answer[1]){
        temp = answer[0];
        answer[0] = answer[1];
        answer[1] = temp;
    }
        
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    w,h = 0,0
    for i in range(1,yellow + 1):
        if yellow % i == 0:
            y_h = i
            y_w = yellow // i
            if 2*y_w + 2*y_h + 4 == brown:
                w,h = y_w + 2, y_h + 2
                break
    if w >= h:
        return [w,h]
    else:
        return [h,w]